modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1235,7 +1231,14 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 
 					j++;
 				}
-				parent_signature += ")";
+				parent_signature += ") -> ";
+
+				if (parent_signature == "null") {
+					parent_signature += "void";
```

Comment:
I'm surprised this even compiles without warnings :)

`parent_signature` obviously can't be null if it's being appended `" ) -> "` just before.

What this code intends to check here is whether `parent_return_type.is_hard_type() ? parent_return_type.to_string() : "Variant"` is `null`.